docs/madr/decisions/018-fault-injection-policy.md

```diff
@@ -0,0 +1,230 @@
+#  Fault Injection policy
+
+- Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4734
+
+## Context and Problem Statement
+
+We want to create a [new policy matching compliant](https://github.com/kumahq/kuma/blob/22c157d4adac7f518b1b49939c7e9ea4d2a1876c/docs/madr/decisions/005-policy-matching.md)
+resource for managing fault injection.
+
+FaultInjection in Kuma is implemented with Envoy's [fault injection
+support](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/fault_filter).
+
+## Considered Options
+
+* Create a MeshFaultInjection
+
+## Decision Outcome
+
+Chosen option: create a MeshFaultInjection
+
+## Decision Drivers
+
+- Replace existing policies with new policy-matching compliant resources
+
+## Solution
+
+### Current configuration
+Below is a sample fault injection configuration.
+
+```yaml
+spec:
+  sources:
+    - match:
+        kuma.io/service: frontend_default_svc_80
+        kuma.io/protocol: http
+  destinations:
+    - match:
+        kuma.io/service: backend_default_svc_80
+        kuma.io/protocol: http
+  conf:
+    abort:
+      httpStatus: 500
+      percentage: 50
+    delay:
+      percentage: 50.5
+      value: 5s
+    responseBandwidth:
+      limit: 50 mbps
+      percentage: 50
+```
+
+The configuration translates to the Envoy configuration in Http's filters:
+
+```yaml
+{
+"name": "envoy.filters.http.fault",
+"typed_config": {
+  "@type": "type.googleapis.com/envoy.extensions.filters.http.fault.v3.HTTPFault",
+  "delay": {
+  "fixed_delay": "5s",
+  "percentage": {
+    "numerator": 505000,
+    "denominator": "TEN_THOUSAND"
+  }
+  },
+  "abort": {
+  "http_status": 500,
+  "percentage": {
+    "numerator": 50
+  }
+  },
+  "headers": [
+  {
+    "name": "x-kuma-tags",
+    "safe_regex_match": {
+    "google_re2": {},
+    "regex": ".*&kuma.io/protocol=[^&]*http[,&].*&kuma.io/service=[^&]*frontend_default_svc_80[,&].*"
+    }
+  }
+  ],
+  "response_rate_limit": {
+  "fixed_limit": {
+    "limit_kbps": "50000"
+  },
+  "percentage": {
+    "numerator": 50
+  }
+  }
+}
+```
+
+### Specification
+
+Fault injection supports only `HTTP` protocol and is configured in `HttpFilter`. We can use the header `x-kuma-tags` to match the specific source. In the case of `MeshHealthCheck` which doesn't set `x-kuma-tags` we can put `x-kuma-tags` or create `MeshFaultInjection` that allows only `Mesh` in `From` section.
+
+#### Top level
+
+Top-level targetRef can have all available kinds:
+```yaml
+targetRef:
+ kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
+ name: ...
+```
+Fault injection is an inbound policy so only `from` should be configured.
```

Comment:
I am not 100% sure about the implementation of `to` but it might be more complicated because we might have to add configuration on the route. Currently, I've added that it might be possible for MeshServiceSubset but because we are identifying by `x-kuma-tags` it might be not easy/possible to add sometimes because we would need cluster matching. In case of an issue, we can just add `Mesh/MeshService`.

Reply:
Ok, this is just an observation, don't think about it too much for this PR :)